python/setup.py

```diff
@@ -227,7 +227,7 @@ def get_packages(self):
 if setup_spec.type == SetupType.RAY:
     pandas_dep = "pandas >= 1.3"
     numpy_dep = "numpy >= 1.20"
-    pyarrow_dep = "pyarrow >= 6.0.1"
```

Comment:
We cannot just do this given Ray as a library should have as minimal restrictions as possible on the versions of other libraries users may want to use.

Why do we need to update this one? If we need to update it, can we find a new lower bound?